src/storable.rs

```diff
@@ -568,40 +571,192 @@ pub(crate) const fn bounds<A: Storable>() -> Bounds {
+        encoded_bytes_to_store >>= 2;
+    }
+
+    // Because encoding and decoding are started on the same
+    // end of the byte, we need to reverse `bytes_to_store_sizes`
+    // to get sizes in order.
+    bytes_to_store_sizes.reverse();
+
+    bytes_to_store_sizes
+}
+
+impl<A, B, C> Storable for (A, B, C)
+where
+    A: Storable,
+    B: Storable,
+    C: Storable,
+{
+    // Touple will be serialized in the following form:
+    // <size_lengths (1B)> <size_a (1-4B)> <a_bytes> <size_b(1-4B)> <b_bytes> <c_bytes>
+    fn to_bytes(&self) -> Cow<[u8]> {
+        let a_bytes = self.0.to_bytes();
+        let a_size = a_bytes.len();
+
+        let b_bytes = self.1.to_bytes();
+        let b_size = b_bytes.len();
+
+        let c_bytes = self.2.to_bytes();
+        let c_size = c_bytes.len();
+
+        let output_size = 1
+            + bytes_to_store_size(a_size)
+            + a_size
+            + bytes_to_store_size(b_size)
+            + b_size
+            + c_size;
+
+        let mut bytes = vec![0; output_size];
+
+        bytes[0] = encode_size_lengths(vec![a_size, b_size]);
+
+        let mut curr_ind = 1;
+
+        encode_size(&mut bytes[curr_ind..], a_size, bytes_to_store_size(a_size));
+        curr_ind += bytes_to_store_size(a_size);
+        bytes[curr_ind..curr_ind + a_size].copy_from_slice(a_bytes.borrow());
+        curr_ind += a_size;
+
+        encode_size(&mut bytes[curr_ind..], b_size, bytes_to_store_size(b_size));
+        curr_ind += bytes_to_store_size(b_size);
+        bytes[curr_ind..curr_ind + b_size].copy_from_slice(b_bytes.borrow());
+        curr_ind += b_size;
+
+        bytes[curr_ind..curr_ind + c_size].copy_from_slice(c_bytes.borrow());
+
+        debug_assert_eq!(curr_ind + c_size, output_size);
+
+        Cow::Owned(bytes)
+    }
+
+    fn from_bytes(bytes: Cow<[u8]>) -> Self {
+        let size_lengths = decode_size_lengths(bytes[0], 2);
+
+        let mut curr_ind = 1;
+
+        let a_size_len = size_lengths[0] as usize;
+        let a_size = decode_size(&bytes[curr_ind..], a_size_len);
+        curr_ind += a_size_len;
+        let a = A::from_bytes(Cow::Borrowed(&bytes[curr_ind..curr_ind + a_size]));
+        curr_ind += a_size;
+
+        let b_size_len = size_lengths[1] as usize;
+        let b_size = decode_size(&bytes[curr_ind..], b_size_len);
+        curr_ind += b_size_len;
+        let b = B::from_bytes(Cow::Borrowed(&bytes[curr_ind..curr_ind + b_size]));
+        curr_ind += b_size;
+
+        let c = C::from_bytes(Cow::Borrowed(&bytes[curr_ind..]));
+
+        (a, b, c)
+    }
+
+    const BOUND: Bound = {
+        match (A::BOUND, B::BOUND, C::BOUND) {
+            (Bound::Bounded { .. }, Bound::Bounded { .. }, Bound::Bounded { .. }) => {
+                let a_bounds = bounds::<A>();
+                let b_bounds = bounds::<B>();
+                let c_bounds = bounds::<C>();
+
+                Bound::Bounded {
+                    max_size: a_bounds.max_size
```

Comment:
Is this `max_size` correct? Aren't we also storing an additional byte for the sizes? Also, we're not storing the size of `C` anymore. We should have test cases that can detect these errors as well.

Reply:
Can we also add tests that also assert the max size to prevent this regression?